### ISSUE TITLE
Support for bgzipped fasta indexes

### ISSUE THREAD

**Issue by magicDGS** [samtools/htsjdk#864]
### Motivation
After #842, there is the possibility to create .fai indexes for a FASTA file, either bgzipped or not. Nevertheless, bgzipped files are not fully supported:

- [x] Index creation for bgzipped FASTA does not creates a .gzi index, which is a difference with respect to samtools. This index cannot be created for other bgzipped files, and will be interesting to be able to do it for any kind of file too.
- [ ] Indexed bgzipped FASTA cannot be used within HTSJDK. This is supported in samtools, but this limits the usage of compressed FASTA in toolkits such as Picard or GATK.

It will be great if this support is implemented.

**Comment by yfarjoun** [comment]
this seems like a nice project....any takers?

**Comment by ffinfo** [comment]
:+1:

**Comment by magicDGS** [comment]
@ffinfo, there is a PR (#997) for the .gzi indexes if you wanna have a look!

**Comment by magicDGS** [comment]
Update: new PR for fully supporting bgzip FASTA references (#1014)

**Comment by magicDGS** [comment]
@ffinfo - the PR is in! Next release HTSJDK will have compressed FASTA support!
